### ISSUE TITLE
12.13.3 User-defined conditional logical operators vs nullable types

### ISSUE THREAD

**Issue by jskeet** [dotnet/csharpstandard#171]
This is taken from ECMA-TC49-TG2/spec#920 but that talks about `Nullable<bool>` as well, which isn't a problem. This issue is just about nullable user-defined types with appropriate operators defined.

---

See also dotnet/roslyn#14098 for examples.

The spec isn't clear what is supposed to happen when a short-circuit operator && or || is used with an expression of a nullable user-defined type. 

---

Basically, either 13.13.1 needs to prohibit `&&` and `||` operating over nullable user-defined types (as it does for `Nullable<bool>`), or 13.13.13 needs to say how it's handled.
